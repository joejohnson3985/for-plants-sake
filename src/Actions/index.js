export const setTrails = trails => ({
  type: 'SET_TRAILS',
  trails
})