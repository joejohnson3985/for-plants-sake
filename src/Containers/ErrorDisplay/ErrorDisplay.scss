@import '../../Utilities/_Variables.scss';

.show-error-bar {
  @include error;
}