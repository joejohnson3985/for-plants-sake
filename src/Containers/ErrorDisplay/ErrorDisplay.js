import React, { Component } from 'react';
import { connect } from 'react-redux';
import './ErrorDisplay.scss';
import { setError } from '../../Actions';


export class ErrorDisplay extends Component {

  clearError = () => {
    let cleared = ''
    setTimeout(() => this.props.setError(cleared), 3500)
  }

  render() {
    let whatToRender;
    if(this.props.error.length) {
      whatToRender = <p>{this.props.error}</p>
      this.clearError()
    }
    return(
      <div className='show-error-bar'>
        {whatToRender}
      </div>
    ) 
  }
}

const mapStateToProps = state => ({
  error: state.error
})

const mapDispatchToProps = dispatch => ({
  setError: error => dispatch(setError(error))
})

export default connect(mapStateToProps, mapDispatchToProps)(ErrorDisplay)