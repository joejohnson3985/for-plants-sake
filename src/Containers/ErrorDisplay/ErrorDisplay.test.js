import { ErrorDisplay, mapStateToProps, mapDispatchToProps } from './ErrorDisplay';
import { shallow } from 'enzyme';
import React from 'react';
import { setError } from '../../Actions';

describe('ErrorDisplay', () => {
  let wrapper;
  let error;
  describe('Component', () => {
    it('Should match the snapshot when there is no error.', () => {
      error = ''
      wrapper = shallow(
        <ErrorDisplay
          error={error}
        />
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('Should match the snapshot when there is an error.', () => {
      error = 'error'
      wrapper = shallow(
        <ErrorDisplay
          error={error}
        />
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('Should clear the error message', () => {
      error = 'error'
      wrapper = shallow(
        <ErrorDisplay
          error={error}
        />
      )
      expect(wrapper.props.error).toEqual(undefined)
    })
  })
})