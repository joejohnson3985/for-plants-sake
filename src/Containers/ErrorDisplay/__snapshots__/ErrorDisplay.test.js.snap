// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDisplay Component Should match the snapshot when there is an error. 1`] = `
<div
  className="show-error-bar"
>
  <p>
    error
  </p>
</div>
`;

exports[`ErrorDisplay Component Should match the snapshot when there is no error. 1`] = `
<div
  className="show-error-bar"
/>
`;
